### ISSUE TITLE
Create Empty/Orphan Branch

### ISSUE THREAD

**Issue by jyxjjj** [desktop/desktop#7929]
We can only create a branch based on current branches.
If we want to create an empty branch
We must use the command line.
Can desktop add this?
```
git checkout --orphan *EmptyBranchName*
```

**Comment by outofambit** [comment]
Hi @jyxjjj and thanks for this feature request. Can you tell us more about your workflow and how/when you use orphan branches? thanks!

**Comment by tierninho** [comment]
When testing this, I spotted two console errors after creating the orphaned branch. Not sure if they need some attention:

```
install.ts:23 `git reflog --date=iso --after="2019-07-04T18:55:30.624Z" --pretty=%H %gd %gs --grep-reflog=checkout: moving from .* to .*$ --` exited with an unexpected code: 128.
stderr:
fatal: your current branch 'adasda' does not have any commits yet

///and///

Uncaught (in promise) GitError: fatal: your current branch 'adasda' does not have any commits yet
```

**Comment by outofambit** [comment]
hmm @tierninho that is a bug we should probably document, but its not a big deal. (it looks like the branch pruner is generating this error). i think that would be easy to fix.

_____

@jyxjjj we haven't heard back from you and this is the first time we've gotten this request, so i'm going to close this issue. i suspect this is not a common workflow, but we can reopen this issue for further feedback and discussion if needed.

**Comment by jyxjjj** [comment]
Sorry, i forgot here is an issue i created and i found it so long later.
My workflow like #4654. And i don't think my request or issue has any connections with #7983.

**Comment by rouleau** [comment]
Just to add a comment, I currently find myself in a situation very similar to previous requests. I need to create a new branch from scratch without the history or files of other branches. You know, start fresh. So, I was hoping to do it with GitHub Desktop or the GitHub website. But apparently you can't?

No big deal, though, since I'll just use the command line to create the new orphaned branch but it would have been nice to use either the website or the desktop.

**Comment by jyxjjj** [comment]
More : I hope add ```reset --hard``` and ```force push``` too.
When i do something to main branch to fix some temp issue.
And i create a branch named ```reverter``` immediately, to delete this fix by revert.
But then i found that fix is not enough, so i did more to main branch.
Then i need to reset --hard to the hash that did not reverted on ```reverter``` branch, and should force push it.
Then cherry-picked more fix from main branch.
Then revert all this fixes push to ```reverter``` branch.

It may like this:
```
Branch 'main':
commit ccc (temp fix 2)
commit bbb (temp fix 1)
commit aaa (needed)
```
```
Branch 'reverter':
reverted commit bbb (to revert fix 1)
commit bbb (temp fix 1)
commit aaa (needed)
```
Then i need make ```reverter``` like:
```
Branch 'reverter':
commit bbb (temp fix 1)
commit aaa (needed)
```
so that i can cherry-pick:
```
Branch 'reverter':
commit ccc (temp fix 2)
commit bbb (temp fix 1)
commit aaa (needed)
```
then revert:

```
Branch 'reverter':
revert commit bbb (to revert temp fix 1)
revert commit ccc (to revert temp fix 2)
commit ccc (temp fix 2)
commit bbb (temp fix 1)
commit aaa (needed)
```

**Comment by andrewhong04** [comment]
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/35881688/218252287-f820caca-0dc6-4956-8f18-e55885fcbbfe.png">

Is this okay?

**Comment by icockrum** [comment]
git checkout --orphan *EmptyBranchName*

fwiw four years later, I also would like to see the workflow described by the original poster added to the desktop app, as my team often wants to create empty branches in which to do exploratory bits of code that may never be merged into the main project.

**Comment by andrewhong04** [comment]
> git checkout --orphan *EmptyBranchName*
> 
> fwiw four years later, I also would like to see the workflow described by the original poster added to the desktop app, as my team often wants to create empty branches in which to do exploratory bits of code that may never be merged into the main project.
> 

I can expand this into a pull request this weekend then.

**Comment by jyxjjj** [comment]
😆😂🤣 They CLOSED THAT #16345  . 🤣🤣🤣
